src/libcore/mem.rs

```diff
@@ -1148,4 +1148,18 @@ impl<T> MaybeUninit<T> {
     pub fn as_mut_ptr(&mut self) -> *mut T {
         unsafe { &mut *self.value as *mut T }
     }
+
+    /// Get a pointer to the first element of the array.
+    #[unstable(feature = "maybe_uninit", issue = "53491")]
+    #[inline(always)]
+    pub fn first_ptr(this: &[MaybeUninit<T>]) -> *const T {
+        this as *const [MaybeUninit<T>] as *const T
+    }
+
+    /// Get a mutable pointer to the first element of the array.
+    #[unstable(feature = "maybe_uninit", issue = "53491")]
+    #[inline(always)]
+    pub fn first_mut_ptr(this: &mut [MaybeUninit<T>]) -> *mut T {
```

Comment:
I called it `mut_ptr` because of `MaybeUninit::as_mut_ptr`.  Should I renamed that as well?

And then there is https://doc.rust-lang.org/nightly/std/primitive.slice.html#method.as_mut_ptr...

Reply:
Uh, ok. I’m fine with landing with either to unblock this PR, we can discuss again before stabilization.